cartography/intel/gcp/dns.py

```diff
@@ -121,6 +121,103 @@ def get_dns_rrs(dns: Resource, dns_zones: List[Dict], project_id: str) -> List[R
             # raise
             return []
 
+@timeit
+def get_dns_keys(dns: Resource, dns_zones: List[Dict], project_id: str) -> List[Resource]:
+    """
+    Returns a list of DNS Keys within the given project.
+
+    :type dns: The GCP DNS resource object
+    :param dns: The DNS resource object created by googleapiclient.discovery.build()
+
+    :type dns_zones: list
+    :param dns_zones: List of DNS zones for the project
+
+    :type project_id: str
+    :param project_id: Current Google Project Id
+
+    :rtype: list
+    :return: List of Resource Record Sets
+    """
+    try:
+        dns_keys = []
+        for zone in dns_zones:
+            request = dns.dnsKeys().list(project=project_id, managedZone=zone['name'])
+            while request is not None:
+                response = request.execute()
+                if response.get('dnsKeys'):
```

Comment:
we should avoid nested loops like this. Instead, return all the keys from this method and create a separate trahsform method to assign zone & id values.

Reply:
There would be transform method for each resource and then some of these methods will also have loops and some will have to populate only one field.